client/my-sites/domains/domain-management/site-redirect/index.jsx

```diff
@@ -3,151 +3,222 @@
 import Card from 'components/card/compact';
 import SectionHeader from 'components/section-header';
-
-const SiteRedirect = React.createClass( {
-	mixins: [ analyticsMixin( 'domainManagement', 'siteRedirect' ) ],
-
-	propTypes: {
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+} from 'state/analytics/actions';
+
+class SiteRedirect extends React.Component {
+	static propTypes = {
 		location: React.PropTypes.object.isRequired,
 		selectedDomainName: React.PropTypes.string.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [
 			React.PropTypes.object,
 			React.PropTypes.bool
 		] ).isRequired
-	},
+	};
+
+	state = {
+		location: this.props.location.value
+	};
 
 	componentWillMount() {
 		upgradesActions.fetchSiteRedirect( this.props.selectedSite.domain );
-	},
+	}
 
-	componentWillReceiveProps: function( nextProps ) {
+	componentWillReceiveProps( nextProps ) {
 		if ( this.props.location.value !== nextProps.location.value ) {
 			this.setState( {
 				location: nextProps.location.value
 			} );
 		}
-	},
+	}
 
 	componentWillUnmount() {
-		upgradesActions.closeSiteRedirectNotice( this.props.selectedSite.domain );
-	},
+		this.closeRedirectNotice();
+	}
 
-	getInitialState() {
-		return {
-			location: this.props.location.value
-		};
-	},
+	closeRedirectNotice = () => {
+		upgradesActions.closeSiteRedirectNotice( this.props.selectedSite.domain );
+	};
 
-	handleChange( event ) {
+	handleChange = ( event ) => {
 		let location = event.target.value;
 
 		// Removes the protocol part
 		location = location.replace( /.*:\/\//, '' );
 
 		this.setState( { location } );
-	},
+	};
 
-	handleClick( event ) {
+	handleClick = ( event ) => {
 		event.preventDefault();
 
 		upgradesActions.updateSiteRedirect( this.props.selectedSite.domain, this.state.location, ( success ) => {
-			this.recordEvent( 'updateSiteRedirectClick', this.props.selectedDomainName, this.state.location, success );
+			this.props.updateSiteRedirectClick(
+				this.props.selectedDomainName,
+				this.state.location,
+				success
+			);
 
 			if ( success ) {
 				page( paths.domainManagementRedirectSettings( this.props.selectedSite.slug, this.state.location ) );
 			}
 		} );
-	},
+	};
 
-	handleFocus() {
-		this.recordEvent( 'locationFocus', this.props.selectedDomainName );
-	},
+	handleFocus = () => {
+		this.props.locationFocus( this.props.selectedDomainName );
+	};
 
 	render() {
-		let classes = classNames(
+		const { location, translate } = this.props;
+		const { isUpdating, notice } = location;
+		const isFetching = location.isFetching || this.state.location === null;
+
+		const classes = classNames(
 			'site-redirect-card',
-			{ fetching: this.props.location.isFetching }
+			{ fetching: isFetching }
 		);
 
 		return (
 			<div>
-				<SiteRedirectNotice
-					notice={ this.props.location.notice }
-					selectedSite={ this.props.selectedSite } />
-
-				<Main className="domain-management-site-redirect">
+				{
+					notice &&
```

Comment:
The notice should probably be inside `Main`, possibly under `Header`, so that it's the same width as the rest of the view. Otherwise it looks a bit off:
<img width="970" alt="screen shot 2017-07-26 at 00 02 56" src="https://user-images.githubusercontent.com/3392497/28595832-fd5dbed2-7195-11e7-9224-72af052bb559.png">

Quick test with the above suggestion:
<img width="991" alt="screen shot 2017-07-26 at 00 07 41" src="https://user-images.githubusercontent.com/3392497/28595951-872f46f8-7196-11e7-920e-0350025ebb64.png">